src/apps/organisations/__test__/components/OrgForm.test.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { Organisation } from "../../types";
+import { render, fireEvent, act } from "@testing-library/react";
+import { BrowserRouter as Router } from "react-router-dom";
+import { CONTEXT } from "../../../../utils";
+import OrganisationForm from "../../components/OrgForm";
+
+type orgFormProps = React.ComponentProps<typeof OrganisationForm>;
+
+let onSubmit = jest.fn();
+
+const testOrg: Organisation = {
+  id: "T1",
+  name: "Test organisation",
+  company: "Test company",
+  website: "https://test.com",
+  location: "Belgium",
+  public_access: "View",
+  extras: null
+};
+const baseProps: orgFormProps = {
+  onSubmit: onSubmit,
+  loading: true,
+  status: "",
+  errors: [],
+  context: CONTEXT.view,
+  savedValues: testOrg
+};
+
+function renderUI(props: Partial<orgFormProps> = {}) {
+  return render(
+    <Router>
+      <OrganisationForm {...baseProps} {...props} />
+    </Router>
+  );
+}
+
+describe("View OrgForm ", () => {
+  it("snapshot test", () => {
+    const { container } = renderUI();
+    expect(container).toMatchSnapshot();
+  });
+  it("should populate mandatory field values", () => {
+    const { getByLabelText } = renderUI();
+    // @ts-ignore
+    expect(getByLabelText("Organisation ID").value).toBe("T1");
+    // @ts-ignore
+    expect(getByLabelText("Organisation Name").value).toBe("Test organisation");
+  });
+  it("should populate non mandatory field values", () => {
+    const { getByLabelText } = renderUI();
+    // @ts-ignore
+    expect(getByLabelText("Company").value).toBe("Test company");
+  });
+});
+
+describe("Create OrgForm", () => {
+  it("should be able to enter values for Organization ID ", async () => {
+    const { getByLabelText, getByText } = renderUI({
+      context: CONTEXT.create
+    });
+    let orgID: HTMLInputElement = getByLabelText(
+      /Organisation ID/
+    ) as HTMLInputElement;
+    let submitButton: HTMLInputElement = getByText(
+      "Submit"
+    ) as HTMLInputElement;
+    await act(async () => {
+      fireEvent.change(orgID, { target: { value: "T1" } });
```

Comment:
What happens if we change the orgID value here to maybe `T2` to show that you are creating a new org different from the test org we used, that way it won't be confusing. The same applies to the below creations.

Reply:
@hadijahkyampeire, the test won't pass as the expected and received values are different.

Reply:
It's because you are passing saved values in baseProps, so it's like these tests are for only viewing a concept, if you pass savedValues as props to only where you are testing the view, and then make them undefined on the baseProps then you can test creating correctly. Let's chat about it.

Reply:
```suggestion
      fireEvent.change(orgID, { target: { value: "Org2" } });
```